test/common/common/utility_test.cc

```diff
@@ -877,27 +877,40 @@ TEST(InlineStorageTest, InlineString) {
 
 TEST(ContainerRemoveElementsTest, Containers) {
   auto l = StringUtil::splitToken("one,two,three", ",");
-  std::function<bool(const absl::string_view&)> onep = [](const absl::string_view& s) {
-    return "one" == s;
+  std::function<bool(absl::string_view)> onep = [](absl::string_view s) { return "one" == s; };
```

Comment:
Oh I think I see the problem. Can you try:
```
auto onep = [](absl::string_view s) -> bool { return "one" == s; };
```

Reply:
nope: `utility.h:726:15: note: candidate template ignored: could not match 'function<bool (type-parameter-0-1)>' against '(lambda at test/common/common/utility_test.cc:881:15)'`

I walked over a lot of the lambda syntax space trying to figure this out yesterday :/.

Reply:
OK I have a marginally better solution, verified by patching your PR.

Let's swap the order of your template args in utility.h:
```
  template <typename T, typename Container>
      static void removeMatchingElements(Container& c, std::function<bool(T)> predicate) {
```

Then at the call-site give the first template arg explicitly:
  Containers::removeMatchingElements<absl::string_view>(l, onep);

IMO that's nicer than having to spell out the lambda type. WDYT?

IDK why it can't infer more, but I imagine the templating inference in the compiler is quite complex.

Reply:
OK I got the epiphany on how to do this right. You don't need to qualify at the call-site; you just need to change to a single-arg template:
```
  template <typename Container> static void removeMatchingElements(
      Container& c, std::function<bool(typename Container::value_type)> predicate) {
```

Reply:
teamwork makes the dream work \o/